Development/nmos-cpp-node/node_implementation.cpp

```diff
@@ -1269,6 +1391,233 @@ nmos::channelmapping_activation_handler make_node_implementation_channelmapping_
+            { nmos::caps::format::color_sampling, nmos::make_caps_string_constraint({ color_sampling }) },
+            { nmos::caps::format::interlace_mode, nmos::make_caps_string_constraint({ interlace_mode.name }) },
+            { nmos::caps::format::colorspace, nmos::make_caps_string_constraint({ colorspace }) },
+            { nmos::caps::format::transfer_characteristic, nmos::make_caps_string_constraint({ transfer_characteristic }) },
+            { nmos::caps::format::component_depth, nmos::make_caps_integer_constraint({ component_depth }) },
+            { nmos::caps::transport::st2110_21_sender_type, nmos::make_caps_string_constraint({ sdp::type_parameters::type_N.name }) }
+        })
+    });
+
+    auto audio_sender_capabilities = value_of({
+        value_of({
+            { nmos::caps::format::media_type, nmos::make_caps_string_constraint({ nmos::media_types::audio_L(24).name }) },
+            { nmos::caps::format::channel_count, nmos::make_caps_integer_constraint({ channel_count }) },
+            { nmos::caps::format::sample_rate, nmos::make_caps_rational_constraint({ nmos::rational{48000, 1} }) },
+            { nmos::caps::format::sample_depth, nmos::make_caps_integer_constraint({ 24 }) }
+        })
+    });
+
+    return [&gate, video_sender_capabilities, audio_sender_capabilities, video_sender_ids, audio_sender_ids](const nmos::resource& streamcompatibility_sender, const web::json::value& active_constraints, web::json::value& intersection)
+    {
+        const auto& sender_id = streamcompatibility_sender.id;
+        const bool video_found = video_sender_ids.end() != boost::range::find(video_sender_ids, sender_id);
+        const bool audio_found = audio_sender_ids.end() != boost::range::find(audio_sender_ids, sender_id);
+
+        const auto& sender_capabilities_ = video_found ? video_sender_capabilities : audio_found ? audio_sender_capabilities : throw std::logic_error("No Sender Capabilities found");
+        const auto& sender_capabilities = sender_capabilities_.as_array();
+        const auto& constraint_sets = nmos::fields::constraint_sets(active_constraints).as_array();
+
+        std::vector<web::json::value> v(constraint_sets.size() * sender_capabilities.size());
+        auto iter = v.begin();
+
+        for (const auto& constraint_set : constraint_sets)
+        {
+            if (!nmos::caps::meta::enabled(constraint_set)) continue;
+            for (const auto& sender_caps_constraint_set : sender_capabilities)
+            {
+                const auto intersection = nmos::experimental::get_constraint_set_intersection(sender_caps_constraint_set, constraint_set, true);
+                if (!intersection.is_null())
+                {
+                    *iter++ = std::move(intersection);
+                }
+            }
+        }
+
+        v.resize(iter - v.begin());
+        if (!v.empty())
+        {
+            intersection = value_from_elements(v);
```

Comment:
I suspect there are cases where there will be some duplicate constraint set intersections in the N*M list. Maybe not very often, so it doesn't matter if we don't remove those for now?

Reply:
`set` with reallocation vs. `vector` with duplicates? Hm, the latter seems more reasonable to me so far.